.clj-kondo/config.edn

```diff
@@ -0,0 +1,10 @@
+{:skip-comments true
+ ;; no linting for all dev/user.clj namespaces
+ :config-in-ns {user {:ignore true}}
+ ;; CMR custom macros
+ :lint-as {cmr.common.config/defconfig clojure.core/def
+           cmr.common.jobs/defjob clojure.core/defn
+           cmr.common.jobs/def-stateful-job clojure.core/defn
+           cmr.common.util/are3 clj-kondo.lint-as/def-catch-all}
+ ;; dynamically generated functions/vars
+ :linters {:unresolved-var {:exclude [digest/md5 digest/sha-1]}}}
```

Comment:
This is much better than the declare statements that I was using.  
Can we also add are2 and are?  They are deprecated functions, but the test code still uses them and are showing up as such from the linters.

Couple of questions 
1) When you merge this in does this work when running kondo only from the top level of the project or can kondo be run inside of each project and it will pick up this file?  Or do you have to tell it about the file?  I am using my own installed cli-kondo executable at the moment.
2) After you merge in and I understand the answer to my first question I am assuming that I should remove the declare statements that I put in?

Reply:
1. Interestingly, I would have thought it would only work from the root level as well, but I just tried it and it does still work in the subprojects. I am using my own clj-kondo executable as well and have not added clj-kondo as a dep to CMR, just this config file since clj-kondo is now our officially supported linter to my knowledge.

I tested this by changing into the common-app-lib directory and running `clj-kondo --lint ./src`, which still suppressed all the stuff mentioned here. To make sure I tried deleting the contents of config.edn and running it again and it did come back.

2. I think that a lot of those, that you previously added, might be covered by what is here, but possibly not all. Maybe we should coordinate on rules for what to consistently ignore/exclude/include etc.

Reply:
And yes, I will add are2 and are3 as well.

Reply:
it is just are2. are3 you already included and are is a clojure command.

Reply:
Note for Reader: discussed with reviewer here and we decided I will review the aforementioned declares that were already merged in for access-control-app, acl-lib, and es-spatial-plugin, and incorporate into this PR.

Reply:
reader has noted discussion.